lib/msf/core/exploit/remote/socket_server.rb

```diff
@@ -166,12 +182,14 @@ def _determine_server_comm(ip, srv_comm = datastore['ListenerComm'].to_s)
       raise(RuntimeError, "Socket Server Comm (Session #{srv_comm}) does not exist") unless comm
       raise(RuntimeError, "Socket Server Comm (Session #{srv_comm}) does not implement Rex::Socket::Comm") unless comm.is_a? ::Rex::Socket::Comm
     when nil, ''
-      comm = Rex::Socket::SwitchBoard.best_comm(ip)
+      unless ip.nil?
+        comm = Rex::Socket::SwitchBoard.best_comm(ip)
+      end
```

Comment:
This avoids a crash if `ip` is nil which is the case in some unit tests. If the IP is nil, and the comm should be automatically determined, it'll default to local.